src/rpc/blockchain.cpp

```diff
@@ -1167,6 +1168,141 @@ static RPCHelpMan pruneblockchain()
     };
 }
 
+static RPCHelpMan dumpcoinstats()
+{
+    return RPCHelpMan{"dumpcoinstats",
+                "\nDumps content of coinstats index to a CSV file.\n",
+                {
+                    {"filename", RPCArg::Type::STR, RPCArg::Optional::NO, "The filename with path (absolute path recommended)"},
+                    {"cummulative", RPCArg::Type::BOOL, RPCArg::Default{false}, "The numbers coming out of coinstatsindex will be cummulative"},
+                },
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",
+                    {
+                        {RPCResult::Type::STR, "filename", "The filename with full absolute path"},
+                    }
+                },
+                RPCExamples{
+                    HelpExampleCli("dumpcoinstats", "\"test\"") +
+                    HelpExampleRpc("dumpcoinstats", "\"test\"")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    if (request.params[0].isNull())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "File name and path for dump file are required");
+
+    fs::path filepath = fs::u8path(request.params[0].get_str());
+    filepath = fs::absolute(filepath);
+
+    if (fs::exists(filepath)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, filepath.u8string() + " already exists. If you are sure this is what you want, move it out of the way first");
+    }
+
+    fsbridge::ofstream file;
+    file.open(filepath);
+    if (!file.is_open())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot open dump file");
+
+    if (!g_coin_stats_index) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Coinstats index is not enabled");
+    }
+
+    const bool cummulative{request.params[1].isNull() ? false : request.params[1].get_bool()};
+
+    file << strprintf("height,"
+                      "bestblock,"
+                      "txouts,"
+                      "bogosize,"
+                      "muhash,"
+                      "total_amount,"
```

Comment:
`cumulative ? "total_amount" : "amount"`

Reply:
I was a bit hesitant about this because it might make some scripting harder if the header is different but I guess it makes sense to change it if the values are different. So done now.